src/pieces/directed_piece.ts

```diff
@@ -0,0 +1,20 @@
+import { BoardVector2d } from "geometry";
+import { Direction, Piece, PieceOpitons } from "pieces";
+
+export class DirectedPiece extends Piece {
+  protected _direction?: Direction;
+
+  public constructor(position: BoardVector2d, playerId: number, board: Board, options: PieceOpitons){
+    super(position, playerId, board, options);
+  }
+
+  public set direction(direction: Direction) {
+    if(this.direction === undefined){
```

Comment:
```suggestion
    if(this.direction === undefined) {
```

Reply:
I think it will be great option to introduce penalty system for no space, new line or too many blank lines. 5 PLN for each :)

Reply:
No. >:(